frontend/src/components/common/ContextMenu.js

```diff
@@ -0,0 +1,234 @@
+      },
+      extraClasses
+    }
+  }
+
+  hide = () => {
+    const { contextHide } = this.props
+    return <li className='rc-hide' onClick={contextHide}>
+      <div className='rc-icon' />Hide until refresh
+        <div className='rc-keyboard'>Ctrl+H</div>
+    </li>
+  }
+
+  remove = () => {
+    const { contextRemove, map, currentUser } = this.props
+    const canEditMap = map && map.authorizeToEdit(currentUser)
+    if (!canEditMap) {
+      return null
+    }
+    return <li className='rc-remove' onClick={contextRemove}>
+      <div className='rc-icon' />Remove from map
+        <div className='rc-keyboard'>Ctrl+M</div>
+    </li>
+  }
+
+  delete = () => {
+    const { contextDelete, map, currentUser } = this.props
+    const canEditMap = map && map.authorizeToEdit(currentUser)
+    if (!canEditMap) {
+      return null
+    }
+    return <li className='rc-delete' onClick={contextDelete}>
+      <div className='rc-icon' />Delete
+      <div className='rc-keyboard'>Ctrl+D</div>
+    </li>
+  }
+
+  center = () => {
+    const { contextCenterOn, contextNode, topicId } = this.props
+    if (!(contextNode && topicId)) {
+      return null
+    }
+    return <li className='rc-center'
+      onClick={() => contextCenterOn(contextNode.id)}>
+      <div className='rc-icon' />Center this topic
+      <div className='rc-keyboard'>Alt+E</div>
+    </li>
+  }
+
+  popout = () => {
+    const { contextPopoutTopic, contextNode } = this.props
+    if (!contextNode) {
+      return null
+    }
+    return <li className='rc-popout'
+      onClick={() => contextPopoutTopic(contextNode.id)}>
+      <div className='rc-icon' />Open in new tab
+    </li>
+  }
+
+  permission = () => {
+    const { currentUser, contextUpdatePermissions } = this.props
+    if (!currentUser) {
+      return null
+    }
+    return <li className='rc-permission'>
+      <div className='rc-icon' />Change permissions
+      <ul>
+        <li className='changeP toCommons'
+          onClick={() => contextUpdatePermissions('commons')}>
+          <div className='rc-perm-icon' />commons
+        </li>
+        <li className='changeP toPublic'
+          onClick={() => contextUpdatePermissions('public')}>
+          <div className='rc-perm-icon' />public
+        </li>
+        <li className='changeP toPrivate'
+          onClick={() => contextUpdatePermissions('private')}>
+          <div className='rc-perm-icon' />private
+        </li>
+      </ul>
+      <div className='expandLi' />
+    </li>
+  }
+
+  metacode = () => {
+    const { metacodeSets, contextOnMetacodeSelect,
+            currentUser, contextNode } = this.props
+    if (!currentUser) {
+      return null
+    }
+    return <li className='rc-metacode'>
+      <div className='rc-icon' />Change metacode
+      <div id='metacodeOptionsWrapper'>
+        <MetacodeSelect
+          onMetacodeSelect={id => {
+            contextOnMetacodeSelect(contextNode && contextNode.id, id)
+          }}
```

Comment:
FYI this is another example of the arrow function that gets stored in memory or something on every render. If we have problems with our right click menu's performance lol we can re-evaluate. No action needed on this comment for now